src/til/ut_til/BitmapTests.cpp

```diff
@@ -0,0 +1,329 @@
+            // Do it this way and not with equality so you can see it in output.
+            if (expected)
+            {
+                VERIFY_IS_TRUE(actual);
+            }
+            else
+            {
+                VERIFY_IS_FALSE(actual);
+            }
+        }
+
+        Log::Comment(L"Reset a rectangle of bits not totally overlapped and check only they were cleared.");
+        // 1 1 0 0       1 1 0 0
+        // 1 1 0 0  --\  1|0 0|0
+        // 1 1 0 0  --/  1|0 0|0
+        // 0 0 0 0       0 0 0 0
+        til::rectangle resetZone{ til::point{ 1, 1 }, til::size{ 2, 2 } };
+        bitmap.reset(resetZone);
+
+        for (auto pt : totalZone)
+        {
+            const auto expected = setZone.contains(pt) && !resetZone.contains(pt);
+            const auto actual = bitmap._bits[totalZone.index_of(pt)];
+            // Do it this way and not with equality so you can see it in output.
+            if (expected)
+            {
+                VERIFY_IS_TRUE(actual);
+            }
+            else
+            {
+                VERIFY_IS_FALSE(actual);
+            }
+        }
+    }
+
+    TEST_METHOD(SetResetExceptions)
+    {
+        til::bitmap map{ til::size{ 4, 4 } };
+        Log::Comment(L"1.) SetPoint out of bounds.");
+        {
+            auto fn = [&]() {
+                map.set(til::point{ 10, 10 });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+
+        Log::Comment(L"2.) SetRectangle out of bounds.");
+        {
+            auto fn = [&]() {
+                map.set(til::rectangle{ til::point{
+                                            2,
+                                            2,
+                                        },
+                                        til::size{ 10, 10 } });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+
+        Log::Comment(L"3.) ResetPoint out of bounds.");
+        {
+            auto fn = [&]() {
+                map.reset(til::point{ 10, 10 });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+
+        Log::Comment(L"4.) ResetRectangle out of bounds.");
+        {
+            auto fn = [&]() {
+                map.reset(til::rectangle{ til::point{
+                                              2,
+                                              2,
+                                          },
+                                          til::size{ 10, 10 } });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+    }
+
+    TEST_METHOD(Resize)

```

Comment:
Could you verify the contents of the bitmap on a resize? i.e.:
- setAll; resize to larger; original size should be true, else false
- setAll; resize to smaller; all should still be set

Reply:
Nnnn this was literally what I wrote as "FUTURE" in one of the comments, but I'll do it.